courses/dl2/imdb.ipynb

```diff
@@ -98,43 +150,190 @@
     "col_names = ['labels','text']"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use a random permutation np array to shuffle the text reviews."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": 125,
+   "execution_count": 15,
    "metadata": {},
    "outputs": [],
    "source": [
-    "np.random.seed(42)\n",
-    "trn_idx = np.random.permutation(len(trn_texts))\n",
-    "val_idx = np.random.permutation(len(val_texts))"
+    "np.random.seed(42) # the random seed ensures that the next 2 random permutations result in the same order.\n",
+    "trn_idx = np.random.permutation(len(trn_texts)) # Create an array of numbers for randomly shuffling training review texts\n",
```

Comment:
Now I've looked further, I think you should remove all the inline comments you've added. Nearly all of them are obvious from the code, and where they're not, it would be better to add a markdown explanation of what's happening.

(NB: in general don't explain in comments or markdown what the numpy or python functions do. There are docs for that already, which people should look up when they need them.)